mathics/builtin/arithmetic.py

```diff
@@ -1861,6 +1861,54 @@ class Factorial(PostfixOperator, _MPMathFunction):
     mpmath_name = "factorial"
 
 
+class Factorial2(PostfixOperator, _MPMathFunction):
+    """
+    <dl>
+      <dt>'Factorial2[$n$]'
+      <dt>'$n$!!'
+      <dd>computes the double factorial of $n$.
+    </dl>
+
+    >> 5!!
+     = 15
+
+    >> Factorial2[-3]
+     = -1
+
+    'Factorial2' also accept irrational numbers:
+    >> N[Pi!!, 6]
+     = 3.35237
+
+    'Factorial2' also accept complex numbers:
+    >> N[I!!, 6]
+     = 2.71713 + 0.279527 I
+
+    'Factorial2' is not defined for even negative numbers.
+    #> (-4)!!
+     : Factorial2 is not defined for even negative numbers.
+     = -4!!
+    """
+
+    operator = "!!"
+    precedence = 610
+    mpmath_name = "fac2"
+    sympy_name = "factorial2"
+
+    messages = {"ndf": "Factorial2 is not defined for even negative numbers."}
+
+    def apply(self, number, evaluation):
+        "Factorial2[number_?NumberQ]"
+
+        if isinstance(number, Integer):
+            if number.to_python() < 0 and number.to_python() % 2 == 0:
+                # even negative number
+                return evaluation.message("Factorial2", "ndf")
+
+            return from_sympy(sympy.factorial2(number.to_sympy()))
+        else:
+            return from_mpmath(mpmath.fac2(number.to_mpmath()))
```

Comment:
Look at numbers/constants.py for how we handle which routine to use when there are several backends  like mpmath and sympy for handling.

Reply:
Ok.